### PR TITLE
(FACT-2901) Refactor linux networking resolver and add unit tests

### PR DESCRIPTION
**Pull request by Filipovici-Andrei**
Changes made to the networking linux resolver:
 - Refactored the resolver.
 - Updated the unit tests.
 - Enabled rubocop for it and updated .rubocop_todo.yaml.

### CODE REVIEW

**Comment by Filipovici-Andrei** [comment]
jenkins please test this on all

**Comment by Filipovici-Andrei** [comment]
jenkins please test this on all

**Comment by Filipovici-Andrei** [comment]
jenkins please test this on all

File: lib/facter/resolvers/networking_linux.rb [hunk 1]
```diff
@@ -240,7 +271,15 @@ def search_internal_leases(interface_name)
           return unless dhcp
 
           dhcp = dhcp.match(/SERVER_ADDRESS=(.*)/)
-          dhcp[1] if dhcp
+          interface_data[:dhcp] = dhcp[1] if dhcp
+        end
+
+        def search_dhcpcd_command(interface_data, interface_name)
```
Comment (oanatmaria) [inline lib/facter/resolvers/networking_linux.rb:277]: This methos should be named `search_with_dhcpcd_command` or `search_dhcp_with_dhcpcd`

File: lib/facter/resolvers/networking_linux.rb [hunk 2]
```diff
@@ -36,103 +36,128 @@ def retrieve_network_info(fact_name)
...
+          elsif Socket.const_defined? :PF_PACKET
+            return if ifaddr.addr.nil?
+
+            search_sockaddr_for_mac(ifaddr)
+          end
+        rescue StandardError => e
+          log.debug("Could not read mac, got #{e}")
+        end
+
+        def search_sockaddr_for_mac(ifaddr)
```
Comment (oanatmaria) [inline lib/facter/resolvers/networking_linux.rb:130]: This should be named `search_mac_for/from_sockaddr`

**Review by oanatmaria — APPROVED** [review]

**Pull request merged**